### DISCUSSION TITLE
[Feature] Different Storage Template for images without album

### DISCUSSION THREAD

**Discussion by wolfmarco** [immich-app/immich#9692] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I like the possibility to have control over the folder structure by defining a Storage Template. Very nice feature. I'm only missing one little thing. I would like to have more control over where assets are stored without an album.

Example Storage Template: {{album}}/{{y}}-{{MM}}-{{dd}}-{{filename}}

- Assets with album ends up at UPLOAD_LOCATION/admin/Album Name/2022-02-03-IMAGE_56437.jpg
- Assets without album ends up at UPLOAD_LOCATION/admin/2022-02-03-IMAGE_56437.jpg

What I don't like about it is that all assets without an album end up in the same directory.

My first idea is to be able to specify another template that only applies if the asset does not have an album assigned.

This would enable you to store
 - assets without albums in the folder structure year/month/filename.ext and 
 - assets with albums in albums/filename.ext.

This is currently not possible.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by shirou93** [comment]
+ I have multiple albums and all is stored on username/year/month.

In case when rsync all assets to other location I don't have human readable album dir tree.

**Reply by wolfmarco** [reply]
You can mitigate the problem a little by using the following (or similar) template:

{{y}}/{{album}}/{{y}}-{{MM}}-{{dd}}-{{filename}}

However, this does not prevent albums from being scattered over different years. But that is my current workaround.

**Comment by bo0tzz** [comment]
#6810
